### ISSUE TITLE
Command vmaf `--reference-vfilter`  scale only works with full actual values (not `-1`)

### ISSUE THREAD

**Issue by stderr-to-devnull** [alexheretic/ab-av1#218]
Reference: 4k
Encode: 540p

Attempting to use scaling argument as documented/given as example in the help for `vmaf`, results in error thrown by `ffmpeg`. Specifically, attempting to use `scale=960:-1`, results in `[Parsed_libvmaf_9 @ 0000029ab55a6dc0] input height must match` error.

If using `scale=960:540`, it works. This only happens when using `vmaf` command. An example with the `-1` argument is also given in the help for `vmaf`.

Error thrown:

```sh
.\ab-av1.exe vmaf --reference-vfilter scale=960:-1:flags=lanczos --reference "reference.mp4" --distorted "distorted.mkv"

Error: ffmpeg vmaf exit code -22

---stderr---
[...]
[Parsed_libvmaf_9 @ 0000029ab55a6dc0] input height must match.
[Parsed_libvmaf_9 @ 0000029ab55a6dc0] Failed to configure input pad on Parsed_libvmaf_9
[fc#0 @ 0000029ab554f0c0] Error reinitializing filters!
[fc#0 @ 0000029ab554f0c0] Task finished with error code: -22 (Invalid argument)
[fc#0 @ 0000029ab554f0c0] Terminating thread with return code -22 (Invalid argument)
```

**Comment by stderr-to-devnull** [comment]
I just tried to perform another comparison (different source vs encode set) and this time it's not working at all, `ffmpeg` keeps complaining about height and width for unexplainable reason. Again, using `--reference-vfilter scale=960:540:flags=lanczos`.

`ffprobe` output of source and encode:

```
Stream #0:0[0x1](und): Video: h264 (High) (avc1 / 0x31637661), yuv420p(tv, bt709/bt709/unknown, progressive), 1920x1080 [SAR 1:1 DAR 16:9], 7612 kb/s, 59.94 fps, 59.94 tbr, 60k tbn (default)

Stream #0:0: Video: hevc (Main 10), yuv420p10le(tv, bt709), 960x540 [SAR 1:1 DAR 16:9], 59.94 fps, 59.94 tbr, 1k tbn (default)
```

So, perfect 16:9 square pixel resolutions, the scaling should not produce any mismatch.

**Comment by alexheretic** [comment]
I agree that I would expect -1 to work here. But all ab-av1 is doing in your usage example is passing the vfilter to ffmpeg process. I can't see any action I can take to improve this.

You could try [debugging](https://github.com/alexheretic/ab-av1?tab=readme-ov-file#debug) to see the ffmpeg call that is generated and run that yourself. With that you could ask upstream why it behaves as it does.

**Comment by stderr-to-devnull** [comment]
Thanks for the debugging suggestion.

For the original issue, I see now that actually the SAR is different for the encode because I chose an option in Handbrake that maximizes the storage resolution to which a non-square pixel `SAR` is applied. This `SAR` discrepancy is caused by the encode "storage" resolution (the actual resolution of the video stream) which is forced to `960x540` which IS `16:9` but the source file resolution is `4096x2160` so not `16:9`.  Thus, the encode has a `SAR` of `16:15` to match the source aspect ratio.

For the second point I mentioned, the error was caused by me because I supplied the source file also for the `--distorted` arg.

Apologies for the confusion, it is a bit more difficult sometimes working from a CLI to prevent mistakes.

 it helped me see that the error was on my part for the second scenario I mentioned, in error I was supplying the source to the `--distorted` arg instead of the encode file.
